src/app/effects/board.effect.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { Actions, Effect } from  '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { Notification, Notifications, NotificationType } from 'ngx-base';
+import { Observable } from 'rxjs';
+import { BoardMapper } from '../models/board.model';
+import { normalizeArray } from '../models/common.model';
+import { spaceSelector } from '../models/space';
+import { BoardService } from '../services/board.service';
+import { AppState } from '../states/app.state';
+import * as BoardActions from './../actions/board.actions';
+
+export type Action = BoardActions.All;
+
+@Injectable()
+export class BoardEffects {
+  constructor(
+    private actions$: Actions,
+    private boardService: BoardService,
+    private store: Store<AppState>,
+    private notifications: Notifications
+  ) {}
+
+  @Effect() getBoards: Observable<Action> = this.actions$
+    .ofType<BoardActions.Get>(BoardActions.GET)
+    .withLatestFrom(this.store.select(spaceSelector))
+    .switchMap(([action, space]) => {
+      const spaceTemplateUrl = space.relationships['space-template'].links.related;
+      return this.boardService.getBoardApiUrl(spaceTemplateUrl);
+    })
+    .switchMap((url) => {
+      return this.boardService.getBoards(url)
+        .map(boards => {
+          const boardmapper = new BoardMapper();
+          return boards.data.map(board => {
+            board.relationships.columns.data.map(column => {
+              return boards.included.filter(col => col.id === column.id);
```

Comment:
This is a O(n^3) operation could be done in O(n)

Normalize the columns part in `included` section and run map on that.